kroxylicious-filters/kroxylicious-record-encryption/src/main/java/io/kroxylicious/filter/encryption/config/RecordEncryptionConfig.java

```diff
@@ -54,9 +62,18 @@ private Integer getExperimentalInt(String property) {
     private Long getExperimentalLong(String property) {
         if (experimental.containsKey(property)) {
             Object value = experimental.get(property);
-            if (value instanceof Number number) {
+            if (value == null) {
+                return null;
+            }
+            else if (value instanceof Number number) {
                 return number.longValue();
             }
+            else if (value instanceof String stringValue) {
```

Comment:
this should probably be removed so we just support one type on the JSON blob, the problem I saw was if you get the datatype wrong in the JSON blob then it was silently returning null before. Also, unit tests!

Reply:
I kinda think its a reasonable fall back option but was wondering about failing an assertion (i.e. `assert true == false`) so it blows up in our unit tests but catches formatting errors in the YAML and does something sensible?